.github/workflows/preview_ci.yml

```diff
@@ -1,69 +1,77 @@
 name: Preview - CI
 on:
-  push:
-    paths-ignore:
-      - '**/*.md'
-      - 'mkdocs.yml'
-      - 'src/docs/**/*'
-    branches: [ main ]
+  workflow_dispatch:
+  schedule:
+    # 4:19 AM UTC every day. A random time to avoid peak times of GitHub Actions.
+    - cron: '19 4 * * *'
 env:
   DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
   DOTNET_CLI_TELEMETRY_OPTOUT: true
 jobs:
   test:
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
     name: Build, Test, Deploy
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ubuntu-latest, windows-latest]
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
+      with:
+        # Necessary to count commits below. We assume that no more than 50 commits are pushed in a day to main.
+        fetch-depth: 50
```

Comment:
Why any number if not 1? I assume a single commit existence should trigger the build? So if any option is passed maybe 1? (I do that in CI builds for benchmarks to speed-up the clones).

Reply:
Actually, it doesn't need to be. I was thinking wrong. `1` is the default anyway.